frontend/src/plugins/impl/CodeEditorPlugin.tsx

```diff
@@ -33,6 +37,7 @@ export class CodeEditorPlugin implements IPlugin<T, Data> {
     minHeight: z.number().optional(),
     maxHeight: z.number().optional(),
     showCopyButton: z.boolean().optional(),
+    debounce: z.optional(z.union([z.boolean(), z.number()])),
```

Comment:
instead of optoional, could you do `z.union([z.boolean(), z.number()]).default(false),`. this way the types from above would reduce from `debounce?: boolean | number;` to `debounce: boolean | number;` and its one less case to think about